audobject/core/object.py

```diff
@@ -46,12 +52,13 @@ def arguments(self) -> typing.Dict[str, typing.Any]:
         """  # noqa: E501
         signature = inspect.signature(self.__init__)
 
-        # if 'kwargs' are allowed, check all members
-        # otherwise only arguments from __init__
-        if 'kwargs' in signature.parameters:
-            names = self.__dict__
-        else:
-            names = [p.name for p in signature.parameters.values()]
+        # regular arguments from __init__
+        names = [p.name for p in signature.parameters.values()
+                 if not p.name == 'kwargs']
+
+        # additional keyword arguments
+        if define.KEYWORD_ARGUMENTS in self.__dict__:
```

Comment:
When is `define.KEYWORD_ARGUMENTS` not in `self.__dict__`, because you set it in `__init__()`?
When we load from a YAML file?

Reply:
It's only in `self.__dict__` if `super.__init__()` is called in the child class, which is not necessarily the case. Unlike in C++ the default constructor of the base class is not automatically called in Python.

Reply:
Ah, good point, thanks.